pkg/log/zap/flags.go

```diff
@@ -91,11 +91,28 @@ type levelValue struct {
 func (v *levelValue) Set(l string) error {
 	v.set = true
 	lower := strings.ToLower(l)
+	var lvl int
 	switch lower {
-	case "debug", "info", "error":
-		return v.level.Set(l)
+	case "debug":
+		lvl = -1
+	case "info":
+		lvl = 0
+	case "error":
+		lvl = 2
+	default:
+		i, err := strconv.Atoi(lower)
+		if err != nil {
+			return fmt.Errorf("invalid log level \"%s\"", l)
+		}
+
+		if i > 0 {
+			lvl = -1 * i
+		} else {
+			lvl = i
+		}
```

Comment:
In the flag description, we say zap-level can be any integer value greater than 0, but here we allow all integers. So I think we probably need to change the flag description or this logic.

**TL;DR: I think we should allow only logr-supported levels (`error`, `info`, `debug`, and non-negative integers), and anything else should result in an error.**

Mixing zap and logr confuses things because zap uses negative numbers for increasing levels of verbosity, whereas [logr specifically forbids negative numbers](https://github.com/go-logr/logr/blob/9fb12b3b21c5415d16ac18dc5cd42c1cfdd40c4e/logr.go#L136-L139). Zap also has levels that aren't supported by logr.

Operator _developers_ are dealing with the logr interface, and are probably more likely to think to use positive numbers with this flag to denote higher verbosity.

Operator _users_ are seeing only the flag name (`--zap-level`), and are probably more likely to think to use negative numbers since that's what zap uses natively.

I feel like allowing both positive and negative numbers could be confusing to both developers and users. There was a [decent amount of discussion](https://github.com/operator-framework/operator-sdk/pull/873#discussion_r251292941) in the original PR about what the acceptable integer range and string values would be.

My initial thought was that we should use zap's convention for the flag values since that's the exposed flag name:
- `error`
- `info`/`0`
- `debug`/`-1`
- `-2`
- `-3`
- ...

The reason I ended up recommending the logr convention in the original PR was that using the zap convention could give the impression that all zap levels (`warn`, `panic`, `fatal`, etc) are supported. And since zapr doesn't fail when you pass it the negative numbers associated with those levels through the logr interface, we'll probably get users trying to do something not supported by logr. 

By sticking with only logr-supported levels, it should prevent situations where an operator developer uses zapr's unsupported negative log levels in his code, and later changes to a different logr implementation that panics with those negative numbers.

Reply:
Updated test and made negative values return error.